Sources/UIKitBackend/KeyboardToolbar.swift

```diff
@@ -0,0 +1,180 @@
+import SwiftCrossUI
+import UIKit
+
+/// An item which can be displayed in a keyboard toolbar. Implementers of this do not have
+/// to implement ``SwiftCrossUI/View``.
+public protocol ToolbarItem {
+    /// Convert the item to a `UIBarButtonItem`, which will be placed in the keyboard toolbar.
+    func asBarButtonItem() -> UIBarButtonItem
+}
+
+@resultBuilder
+public enum ToolbarBuilder {
+    public typealias Component = [any ToolbarItem]
+
+    public static func buildExpression(_ expression: some ToolbarItem) -> Component {
+        [expression]
+    }
+
+    public static func buildExpression(_ expression: any ToolbarItem) -> Component {
+        [expression]
+    }
+
+    public static func buildBlock(_ components: Component...) -> Component {
+        components.flatMap { $0 }
+    }
+
+    public static func buildArray(_ components: [Component]) -> Component {
+        components.flatMap { $0 }
+    }
+
+    public static func buildOptional(_ component: Component?) -> Component {
+        component ?? []
+    }
+
+    public static func buildEither(first component: Component) -> Component {
+        component
+    }
+
+    public static func buildEither(second component: Component) -> Component {
+        component
+    }
+
+    public static func buildFinalResult(_ component: Component) -> [UIBarButtonItem] {
```

Comment:
The final result should be `[Component]` if possible so that we can hide the `UIBarButtonItem` implementation detail from users that aren't implementing custom items (reducing the risk of breaking changes if we generalise this API for android at some point).

Reply:
Given that `Component` is already an array it would be easiest to just remove `buildFinalResult` entirely.